From 590dd46a909773f0b3f2754ec15487745afbdd91 Mon Sep 17 00:00:00 2001
Message-ID: <590dd46a909773f0b3f2754ec15487745afbdd91.1708115863.git.mjg@fedoraproject.org>
In-Reply-To: <1932a672db047da3204a445880007fcc522fa7d7.1708115863.git.mjg@fedoraproject.org>
References: <1932a672db047da3204a445880007fcc522fa7d7.1708115863.git.mjg@fedoraproject.org>
From: Julian Smith <julian.smith@artifex.com>
Date: Wed, 3 Jan 2024 16:42:17 +0000
Subject: [PATCH 08/11] Makefile: fix build failures with library soft-links.

We need to use `ln -f` when creating shared library soft-links, otherwise we
fail if they already exist.

(cherry picked from commit 032af8acbfdfcfaf7d7fa63df7f22dacf6b4f630)
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 55fda4854..f8a8dfaa9 100644
--- a/Makefile
+++ b/Makefile
@@ -621,14 +621,14 @@ install-shared-c: install-shared-check shared install-headers
 	install -d $(DESTDIR)$(libdir)
 	install -m 644 $(OUT)/libmupdf.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/
 ifneq ($(OS),OpenBSD)
-	ln -s libmupdf.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/libmupdf.$(SO)
+	ln -sf libmupdf.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/libmupdf.$(SO)
 endif
 
 install-shared-c++: install-shared-c c++
 	install -m 644 platform/c++/include/mupdf/*.h $(DESTDIR)$(incdir)/mupdf
 	install -m 644 $(OUT)/libmupdfcpp.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/
 ifneq ($(OS),OpenBSD)
-	ln -s libmupdfcpp.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/libmupdfcpp.$(SO)
+	ln -sf libmupdfcpp.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/libmupdfcpp.$(SO)
 endif
 
 install-shared-python: install-shared-c++ python
-- 
2.44.0.rc1.222.g52f20dec8d

